src/ucs/debug/log.c

```diff
@@ -120,90 +120,85 @@ size_t ucs_log_get_buffer_size()
 
 static void ucs_log_get_file_name(char *log_file_name, size_t max, unsigned idx)
 {
-    ucs_assert(idx <= ucs_global_opts.log_file_backup_count);
+    ucs_assert(idx <= ucs_global_opts.log_file_rotate);
 
     if (idx == 0) {
-        ucs_strncpy_zero(log_file_name, ucs_log_file_names[0], max);
+        ucs_strncpy_zero(log_file_name, ucs_log_file_name, max);
         return;
     }
 
     ucs_snprintf_zero(log_file_name, max, "%s.%u",
-                      ucs_log_file_names[idx], idx);
+                      ucs_log_file_name, idx);
 }
 
-static void ucs_log_handle_file_max_size(int log_entry_len)
+static void ucs_log_file_rotate()
 {
-    const char *next_token;
     char old_log_file_name[UCS_LOG_MAX_FILE_NAME];
     char new_log_file_name[UCS_LOG_MAX_FILE_NAME];
     unsigned idx, start_idx;
     int ret;
 
-    /* check if it is necessary to find a new storage for logs */
-    if ((log_entry_len + ftell(ucs_log_file)) >=
-        ucs_global_opts.log_file_max_size) {
-        fclose(ucs_log_file);
+    if (ucs_log_file_last_idx == ucs_global_opts.log_file_rotate) {
+        /* remove the last file and log rotation from the
+         * `log_file_rotate - 1` file */
+        ucs_log_get_file_name(old_log_file_name,
+                              UCS_LOG_MAX_FILE_NAME,
+                              ucs_log_file_last_idx);
+        unlink(old_log_file_name);
+        start_idx = ucs_log_file_last_idx - 1;
+    } else {
+        start_idx = ucs_log_file_last_idx;
+        ucs_log_file_last_idx++;
+    }
 
-        if (ucs_global_opts.log_file_backup_count != 0) {
-            if (ucs_log_file_last_idx == ucs_global_opts.log_file_backup_count) {
-                /* remove the last file and log rotation from the
-                 * `log_file_backup_count - 1` file */
-                ucs_log_get_file_name(old_log_file_name,
-                                      UCS_LOG_MAX_FILE_NAME,
-                                      ucs_log_file_last_idx);
-                unlink(old_log_file_name);
-                start_idx = ucs_log_file_last_idx - 1;
-            } else {
-                start_idx = ucs_log_file_last_idx;
-                ucs_log_file_last_idx++;
-            }
+    ucs_assert(ucs_log_file_last_idx <= ucs_global_opts.log_file_rotate);
+    ucs_assert(start_idx <= (ucs_global_opts.log_file_rotate - 1));
 
-            ucs_assert(ucs_log_file_last_idx <=
-                       ucs_global_opts.log_file_backup_count);
-            ucs_assert(start_idx <= (ucs_global_opts.log_file_backup_count - 1));
+    for (idx = start_idx; idx != UINT_MAX; --idx) {
```

Comment:
i think it's better to make "idx" a signed integer, and not rely on integer overflow

Reply:
done